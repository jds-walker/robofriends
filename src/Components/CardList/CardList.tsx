import React from "react";
import { IRobot } from "../../App";
import Card from "../Card/Card";

const CardList = ({ robots }: { robots: Array<IRobot> }) => {
  return (
    <div>
      <Card
        robotEmail={robots[0].email}
        robotName={robots[0].name}
        robotId={robots[0].id}
      />
    </div>
  );
};

export default CardList;
