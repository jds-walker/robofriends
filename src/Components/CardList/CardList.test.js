import { shallow } from "enzyme";
import React from "react";
import CardList from "./CardList";

it("cardlist snapshot", () => {
  const mockRobots = [
    {
      name: "JohnJohn",
      email: "jon@gmail.com",
      id: 17
    }
  ];
  expect(shallow(<CardList robots={mockRobots} />)).toMatchSnapshot();
});
