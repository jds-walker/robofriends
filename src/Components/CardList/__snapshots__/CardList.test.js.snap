// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cardlist snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardList
    robots={
      Array [
        Object {
          "email": "jon@gmail.com",
          "id": 17,
          "name": "JohnJohn",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Card
          robotEmail="jon@gmail.com"
          robotId={17}
          robotName="JohnJohn"
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "17",
        "nodeType": "function",
        "props": Object {
          "robotEmail": "jon@gmail.com",
          "robotId": 17,
          "robotName": "JohnJohn",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Card
            robotEmail="jon@gmail.com"
            robotId={17}
            robotName="JohnJohn"
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "17",
          "nodeType": "function",
          "props": Object {
            "robotEmail": "jon@gmail.com",
            "robotId": 17,
            "robotName": "JohnJohn",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
